subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectContainer.java

```diff
@@ -84,8 +92,11 @@ public interface NamedDomainObjectContainer<T> extends NamedDomainObjectSet<T>,
 
     /**
      * Defines a new object, which will be created and configured when it is required. An object is 'required' when the object is located using query methods such as {@link NamedDomainObjectCollection#getByName(java.lang.String)} or when {@link Provider#get()} is called on the return value of this method.
-     *
-     * <p>It is generally more efficient to use this method instead of {@link #create(java.lang.String, org.gradle.api.Action)} or {@link #create(java.lang.String)}, as those methods will eagerly create and configure the object, regardless of whether that object is required for the current build or not. This method, on the other hand, will defer creation and configuration until required.</p>
+     * <p>
+     * It is generally more efficient to use this method instead of {@link #create(java.lang.String, org.gradle.api.Action)} or {@link #create(java.lang.String)}, as those methods will eagerly create and configure the object, regardless of whether that object is required for the current build or not.
+     * This method, on the other hand, will defer creation and configuration until required.
+     * <p>
+     * This operation is lazy, the element is NOT realized before being returned wrapped in a {@link NamedDomainObjectProvider provider}.
```

Comment:
```suggestion
     * This operation is lazy, the returned element is NOT a realized one, but the input wrapped in a {@link NamedDomainObjectProvider provider}.
```

Reply:
I don't understand the proposal tbh. "but the input wrapper in" - the word input trips me up. What is returned is not any of the methods inputs.

Maybe:
> This operation is lazy, the returned element is NOT realized. A {@link NamedDomainObjectProvider lazy wrapper} is returned, allowing to continue to use it with other lazy APIs.

Reply:
+1